packages/orbit-components/src/InputField/index.js

```diff
@@ -344,17 +344,10 @@ const InputField: React.AbstractComponent<Props, HTMLInputElement> = React.forwa
     dataAttrs,
   } = props;
 
-  const forID = React.useMemo(() => id || (label ? randomID("inputFieldID") : undefined), [
```

Comment:
What was the purpose of this additional `label ? randomID("inputFieldID") : undefined` logic? Probably nothing important, but did you leave it out of the new implementation on purpose?

Reply:
[it's just about labels.](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label#attr-for) If the label is present, it should have `for` property and `Input` should have that `id`.